pkg/cli/cmd/app/delete/delete.go

```diff
@@ -160,38 +164,48 @@ func (r *Runner) Run(ctx context.Context) error {
 		return err
 	}
 
-	app, err := client.GetApplication(ctx, r.ApplicationName)
-	if clients.Is404Error(err) {
-		r.Output.LogInfo("Application '%s' does not exist or has already been deleted.", r.ApplicationName)
-		return nil
-	} else if err != nil {
-		return err
-	}
+	if !r.Confirm {
+		app, err := client.GetApplication(ctx, r.ApplicationName)
+		if clients.Is404Error(err) {
+			r.Output.LogInfo("Application '%s' does not exist or has already been deleted.", r.ApplicationName)
+			return nil
+		} else if err != nil {
+			return err
+		}
 
-	environmentID, err := resources.ParseResource(*app.Properties.Environment)
-	if err != nil {
-		return err
-	}
+		environmentID, err := resources.ParseResource(*app.Properties.Environment)
+		if err != nil {
+			return err
+		}
 
-	if !r.Confirm {
 		confirmed, err := prompt.YesOrNoPrompt(fmt.Sprintf(deleteConfirmation, r.ApplicationName, environmentID.Name()), prompt.ConfirmNo, r.InputPrompter)
 		if err != nil {
 			return err
 		}
 		if !confirmed {
 			return nil
 		}
+		r.EnvironmentName = environmentID.Name()
 	}
 
-	deleted, err := client.DeleteApplication(ctx, r.ApplicationName)
+	progressText := fmt.Sprintf("Deleting application '%s' from environment '%s'...", r.ApplicationName, r.EnvironmentName)
+
+	deleted, err := r.Delete.DeleteApplicationWithProgress(ctx, client, clients.DeleteOptions{
+		ApplicationNameOrID: r.ApplicationName,
+		ProgressText:        progressText,
+	})
 	if err != nil {
```

Comment:
The error handling uses `strings.Contains(err.Error(), "not found")` which is fragile. Consider checking for specific error types instead of string matching.